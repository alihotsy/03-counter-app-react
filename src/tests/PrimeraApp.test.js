import React from 'react';
// import { render } from '@testing-library/react'
import FirstApp from '../PrimeraApp'
import {shallow} from 'enzyme';

describe('Pruebas en <FirstApp />', () => {
    // test('Debe de mostrar el mensaje "Hola, soy Terry!"', () => {
    //     const saludo = 'Hola, soy Terry!'
    //     const {getByText} = render(<FirstApp saludo = {saludo} />)
    //     expect(getByText(saludo +"@")).toBeInTheDocument();
    // })
    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
         const saludo = 'Hola, soy Terry!';
         const wrapper = shallow(<FirstApp saludo = {saludo} />);
         expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar el subtítulo enviado por props', () => {
        const saludo = 'Hola, soy Terry!';
        const subtitle = 'Soy un subtítulo';
        const wrapper = shallow(<FirstApp saludo = {saludo} subtitle = {subtitle} />);

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitle)
    })
    
    
})
