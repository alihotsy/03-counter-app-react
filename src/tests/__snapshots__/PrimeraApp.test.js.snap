// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Terry!
    !!! 
  </h1>
  <p>
    Soy un valor por defecto
  </p>
</Fragment>
`;
